kasa/protocol.py

```diff
@@ -81,14 +118,18 @@ async def query(host: str, request: Union[str, Dict], retry_count: int = 3) -> D
 
                 _LOGGER.debug("Unable to query the device, retrying: %s", ex)
 
-            finally:
-                if writer:
-                    writer.close()
-                    await writer.wait_closed()
-
         # make mypy happy, this should never be reached..
+        await self._close()
         raise SmartDeviceException("Query reached somehow to unreachable")
```

Comment:
This got somehow reached on a non-plugged in device from homeassistant's setup, but I'm not sure why..

Reply:
I fixed that.  It couldn't connect 3x.  It now raises